docs/site/Loopback-component-authorization.md

```diff
@@ -8,543 +8,227 @@ permalink: /doc/en/lb4/Loopback-component-authorization.html
-    'userAuthorization.actions.userPermissions',
-  );
-}
+    - The interceptor checks to see if the endpoint is annotated with an
+      authorization specification.
+    - It executes all classes tagged as `Authorizer` to enforce access/privilege
+      control.
+    - Based on the result of every `Authorizer` it decides if the current
+      identity has access to the endpoint.
 
-/**
- * Metadata accessor key for authorize method decorator
- */
-export const AUTHORIZATION_METADATA_ACCESSOR = MetadataAccessor.create<
-  AuthorizationMetadata,
-  MethodDecorator
->('userAuthorization.accessor.operationMetadata');
-```
+  - The component also declares various
+    [types](https://github.com/strongloop/loopback-next/blob/master/packages/authorization/src/types.ts)
+    to use in defining necessary classes and inputs by developers.
 
-Now, we need to create three providers
+    - `Authorizer`: A class implementing access policies. Accepts
+      `AuthorizationContext` and `AuthorizationMetadata` as input and returns an
+      `AuthorizationDecision`.
 
-- **_AuthorizationMetadataProvider_** - This will read the decorator metadata
-  from the controller methods wherever the decorator is used.
-- **_AuthorizeActionProvider_** - This holds the business logic for access
-  validation of the user based upon access permissions allowed at method level
-  via decorator metadata above.
-- **_UserPermissionsProvider_** - This is where we can add our business logic to
-  read and unify permissions associated to user via role, with those associated
-  directly to the user. In our case, an explicit allow/deny at user level takes
-  precendence over role permissions. But this business logic may vary
-  apllication to application. So, feel free to customize.
+    - `AuthorizationDecision`: expected type to be returned by an `Authorizer`
 
-{% include code-caption.html content="/src/authorization/providers/authorization-metadata.provider.ts" %}
+    - `AuthorizationMetadata`: expected type of the authorization spec passed to
+      the decorator used to annotate a controller method. Also provided as input
+      parameter to the `Authorizer`.
 
-```ts
-import {
-  Constructor,
-  inject,
-  MetadataInspector,
-  Provider,
-} from '@loopback/context';
-import {CoreBindings} from '@loopback/core';
-
-import {AUTHORIZATION_METADATA_ACCESSOR} from '../keys';
-import {AuthorizationMetadata} from '../types';
-
-export class AuthorizationMetadataProvider
-  implements Provider<AuthorizationMetadata | undefined> {
-  constructor(
-    @inject(CoreBindings.CONTROLLER_CLASS)
-    private readonly controllerClass: Constructor<{}>,
-    @inject(CoreBindings.CONTROLLER_METHOD_NAME)
-    private readonly methodName: string,
-  ) {}
-
-  value(): AuthorizationMetadata | undefined {
-    return getAuthorizeMetadata(this.controllerClass, this.methodName);
-  }
-}
+    - `AuthorizationContext`: contains current principal invoking an endpoint
```

Comment:
one more thing I feel worth mentioning here: the request context is also include din `AuthorizationContext`

Reply:
agreed